charts/global.d.ts

```diff
@@ -0,0 +1,12 @@
+// navigator.share API: taken from https://github.com/Microsoft/TypeScript/issues/18642#issuecomment-576763015
+type ShareData = {
+    title?: string
+    text?: string
+    url?: string
+    files?: ReadonlyArray<File>
+}
+
+interface Navigator {
+    share?: (data?: ShareData) => Promise<void>
+    canShare?: (data?: ShareData) => boolean
+}
```

Comment:
We need this so we can call the API without TypeScript errors. While we don't need to have `canShare` in there, I guess they go together well.

I don't know exactly where to put such a file since there doesn't seem another with such definitions, so I made a new one. TypeScript will probably include this type definition sometime in the future, making this obsolete.

Reply:
Sounds good @MarcelGerber! We don't have a globals file (as far as I am aware).

Why not use `navigator.canShare()` instead of `"share" in navigator` to enable the menu though?

Reply:
That's a little weird because the support for `navigator.share` and `navigator.canShare` is different: iOS Safari supports `share` but not `canShare`. (https://caniuse.com/#feat=mdn-api_navigator_canshare)

Apparently `canShare` is meant for the case where a website wants to share a file with the OS.